File: abc143/abc143_b.cs
// https://atcoder.jp/contests/abc143/submissions/8107486

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var N = int.Parse(Console.ReadLine());
            var d = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

            var result = 0;
            for (int i = 0; i < N; i++)
            {
                for (int j = i + 1; j < N; j++)
                {
                    result += d[i] * d[j];
                }
            }
            Console.WriteLine(result);
        }
    }
}

File: abc/abc143/abc143_b.fs
// https://atcoder.jp/contests/abc143/submissions/12054197

let N = stdin.ReadLine() |> int
let d = stdin.ReadLine().Split() |> Array.map int

let rec solver r i =
    if i = N then r
    else
        let rec loop r j =
            if j = N then solver r (i + 1)
            else loop (r + d.[i] * d.[j]) (j + 1)
        loop r (i + 1)

solver 0 0 |> stdout.WriteLine
